vtrace/platforms/gdbstub.py

```diff
@@ -876,8 +1001,14 @@ def gdbGetRegister(self, regidx):
         logger.debug('Requesting register %s', reg)
         cmd = b"p" + reg
         res = self._msgExchange(cmd)
-        if res[0:1] == b'E':
-            raise Exception('Error occurred while dumping register info: %s' 
+        if not res:
+            regname = self._gdb_reg_fmt[regidx][0]
+            logger.debug('Cannot retrieve register %d (%s) individually' %
+                    (regidx, regname))
+            regs = self.gdbGetRegisters()
+            return regs[regname]
```

Comment:
while i like this thinking personally, is this not overkill?  should a single-register request turn into requesting all registers automatically?  should that be a setting?  or should that be left up to the next layer up?  this is `gdbGetRegister` which should simply implement the "gdb get register" feature cleanly and atomically. 

also, what does `if not res` mean?  if it's None or 0, or an empty string....  what would cause that from `_msgExchange()`?  is this a fix for QEMU things?  i'm a bit confused on this one.

Reply:
`res` is the result returned on the previous line
```
        res = self._msgExchange(cmd)
```
When I was testing this earlier in the year against different servers I found one didn't let me request a single register at a time.  I believe it was the older powerpc gdb server when requesting some of the special registers. It seemed to only support reading the standard registers individually. I looked at what the normal gdb clients did and they solved this problem by requesting all the registers.

I'm not sure why I did `if not res` because it should instead be checking for "if the first character of the response is `E`". I will fix that now.